webpack/assets/javascripts/react_app/components/charts/StatisticsChartBox.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import helpers from '../../common/helpers';
+import { STATUS } from '../../constants';
+import Chart from './Chart';
+import ChartModal from './ChartModal';
+import Loader from '../common/Loader';
+import Panel from '../common/Panel/Panel';
+import PanelHeading from '../common/Panel/PanelHeading';
+import PanelTitle from '../common/Panel/PanelTitle';
+import PanelBody from '../common/Panel/PanelBody';
+import StatisticsStore from '../../stores/StatisticsStore';
+import StatisticsChartActions from '../../actions/StatiscticChartActions';
+import statisticsPage from '../../../pages/statistics_page';
+import styles from './StatisticsChartsListStyles';
+import MessageBox from '../common/MessageBox';
+
+export default class StatisticsChartBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { showModal: false, status: STATUS.PENDING };
+    helpers.bindMethods(this, [
+      'drawChart',
+      'onChange',
+      'onError',
+      'onClick',
+      'closeModal',
+      'openModal',
+      'drawModal']
+    );
+  }
+
+  componentDidMount() {
+    StatisticsChartActions.getStatisticsData(this.props.url);
+    StatisticsStore.addChangeListener(this.onChange);
+    StatisticsStore.addErrorListener(this.onError);
+  }
+
+  componentWillUnmount() {
+    StatisticsStore.removeChangeListener(this.onChange);
+    StatisticsStore.removeErrorListener(this.onError);
+  }
+
+  onChange(event) {
+    if (event.id === this.props.id) {
+      const statistics = StatisticsStore.getStatisticsData(this.props.id);
+
+      this.setState({
+        status: STATUS.RESOLVED,
+        hasData: !!statistics.data.length,
+        data: statistics.data
+      });
+    }
+  }
+
+  onError(info) {
+    const xhr = info.jqXHR;
+    const id = xhr.originalRequestOptions.url.split('/')[1];
+
+    if (id === this.props.id) {
+      this.setState({
+        status: STATUS.ERROR,
+        errorText: info.errorThrown
+      });
+    }
+  }
+
+  onClick() {
+    if (this.state.data && this.state.hasData) {
+      this.openModal();
+    }
+  }
+
+  componentDidUpdate() {
+    this.drawChart();
+  }
+
+  drawChart() {
+    statisticsPage.generateChart(this.props, this.state.data);
+  }
+
+  openModal() {
+    this.setState({ showModal: true });
+  }
+
+  closeModal() {
+    this.setState({ showModal: false });
+  }
+
+  drawModal() {
+    statisticsPage.generateModalChart(this.props, this.state.data);
+  }
+
+  render() {
+    const tooltip = {
+      onClick: this.onClick,
+      title: _('Expand the chart').toString(),
+      'data-toggle': 'tooltip',
+      'data-placement': 'top'
+    };
+
+    const chart = (<Chart {...this.props} key="0"
```

Comment:
I'm in mixed feelings if this actually improve readability or not, when reading it, I was trying to keep finding where it was used assuming there are multiple places it get picked up by, but if its just one usage, I think i rather see it where it actually used.